Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -254,11 +254,11 @@ extension TopTabsViewController: UICollectionViewDataSource {
         tabCell.titleText.text = tab.displayTitle
         
         if tab.displayTitle.isEmpty {
-            if (tab.url?.baseDomain()?.contains("localhost") ?? true) {
+            if (tab.webView?.URL?.baseDomain()?.contains("localhost") ?? true) {
                 tabCell.titleText.text = AppMenuConfiguration.NewTabTitleString
             }
             else {
-                tabCell.titleText.text = tab.displayURL?.absoluteString
+                tabCell.titleText.text = tab.webView?.URL?.absoluteDisplayString()
```

Comment:
Nit: Are these changes needed? If not its better to keep the changes related to the PR/bug.